v2/controllers/egress_controller.go

```diff
@@ -113,9 +113,16 @@ func selectorLabels(name string) map[string]string {
 }
 
 func (r *EgressReconciler) reconcilePodTemplate(eg *coilv2.Egress, depl *appsv1.Deployment) {
+	tmpMap := make(map[string]string)
+	for k, v := range depl.Spec.Template.Annotations {
+		tmpMap[k] = v
+	}
 	target := &depl.Spec.Template
 	target.Labels = make(map[string]string)
 	target.Annotations = make(map[string]string)
+	for k, v := range tmpMap {
```

Comment:
Why did you create `tmpMap`? For me, this seems no-op.
Just removing line 118 might be sufficient.
> target.Annotations = make(map[string]string)

Reply:
I thought that there is a possibility of updating `target.Annoctations` from the state of given as an argument because `target.Annotations` shares a pointer with `depl.Spec.Template.Annotations`.
Therefore I fixed to  copy `depl.Spec.Template.Annotations` to `tmpMap`.

Reply:
I don't understand the problem.

Reply:
When another processes update `depl.Spec.Template.Annotations` at the same time, a `restartedAt` annotation may be lost unexpectedly. 
Therefore I used `tmpMap` to temporarily save the contents, just in case.

I think that such timing is very rare.
So, as you suggested, there is no problem not to use tmpMap and only to delete line 118.

Reply:
If there are multiple goroutines to access the same data, we _must_ use `sync.Mutex` or similar concurrency idioms.
Swapping pointer does not guarantee data-race safeness.

Reply:
Yes, I should have considered to use `sync.Mutex`. 
And I checked that other goroutines access to its map and there is no such routine.
Therefore I'll fix not to use `tmpMap` and to delete line 118.

Reply:
I mean, there are no (and should not be) other goroutines that share `depl` and modify it.

Reply:
Thank you for your confirmation!